src/solvers/refinement/string_refinement.cpp

```diff
@@ -0,0 +1,869 @@
+/*******************************************************************\
+
+Module: String support via creating string constraints and progressively
+        instantiating the universal constraints as needed.
+	The procedure is described in the PASS paper at HVC'13.
+
+Author: Alberto Griggio, alberto.griggio@gmail.com
+
+\*******************************************************************/
+
+#include <ansi-c/string_constant.h>
+#include <util/replace_expr.h>
+#include <solvers/sat/satcheck.h>
+#include <sstream>
+#include <solvers/refinement/string_refinement.h>
+#include <langapi/language_util.h>
+
+string_refinementt::string_refinementt(const namespacet &_ns, propt &_prop):
+  supert(_ns, _prop)
+{
+  use_counter_example=false;
+  variable_with_multiple_occurence_in_index=false;
+  initial_loop_bound=100;
```

Comment:
I added this as an option to the constructor with default set to 100. https://github.com/diffblue/cbmc/pull/374/commits/b1df7606b160410529f6c922af332a963b46a611
This bound is to ensure that the algorithm terminate but there is no 'right' value for it.